pages/checkout/[id].vue

```diff
@@ -0,0 +1,92 @@
+<template>
+  <h1 class="mb-11">Checkout</h1>
+  <div class="mb-80 grid grid-cols-2 gap-x-72">
+    <CheckoutDetails
+      @update:date="handleDate"
+      @update:days="handleDays"
+      @update:insurance="handleInsurance"
+      @update:license="handleLicense"
+    />
+    <CheckoutSummary
+      :product="selectedProduct"
+      :order="formData"
+    />
+    <div class="flex justify-between">
+      <BaseButton
+        arrow="left"
+        variant="text"
+      >
+        Back
+      </BaseButton>
+      <BaseButton
+        arrow="right"
+        :disabled="!formValid"
+        @click.prevent="handleSubmit"
+      >
+        Book
+      </BaseButton>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { Order } from '~/types/order';
+import { RadioOption } from '~/types/radio-options';
+
+const { data: userProfile } = useUserProfile();
+const { mutate: updateLicense } = useUserProfileUpdate();
+const { mutate: insertReservation, isSuccess: reservationSuccess } =
+  useInsertReservation();
+
+const hasLicense = computed(() => {
+  const currentLicense = userProfile.value?.current_license?.split('/')[1];
+  return currentLicense || null;
+});
+
+const { data: selectedProduct } = useProductById(useRoute().params.id);
+
+const formData: Order = reactive({
+  user_id: userProfile.value?.id || null,
+  date: new Date().toLocaleDateString() || null,
+  duration: 1,
+  license: null,
+  insurance: null,
+});
+
+const handleDate = (e: string) => (formData.date = e);
+const handleDays = (e: number) => (formData.duration = e);
+const handleInsurance = (e: RadioOption) => (formData.insurance = e);
+const handleLicense = (e: File) => (formData.license = e);
+
+const formValid = computed(() => {
+  return !!(
+    formData.user_id &&
+    formData.date &&
+    formData.duration &&
+    (hasLicense || formData.license) &&
+    formData.insurance
+  );
+});
+
+const handleSubmit = async () => {
+  if (formValid.value) {
+    const orderData = {
+      ...formData,
+      prod_id: selectedProduct.value?.id,
+      insurance: formData.insurance?.id,
+      status: 'pending',
+      title: selectedProduct.value?.name,
+    };
+    // console.log(orderData);
+    if (formData.license?.name)
+      await updateLicense({
+        profileName: userProfile.value?.id,
+        file: formData.license,
+      });
+    await insertReservation(orderData);
```

Comment:
These two `await` keywords have no effect. `mutate` methods are synchronous. If you want to use `await` here, you must get from your composables `mutateAsync` instead.
```ts
const { mutateAsync: updateLicense } = useUserProfileUpdate();
const { mutateAsync: insertReservation, isSuccess: reservationSuccess } =
  useInsertReservation(); 
```
Or you can also use mutation lifecycle hooks:
https://vue-query.vercel.app/#/guides/mutations?id=mutation-side-effects